src/lib.rs

```diff
@@ -236,3 +275,95 @@ pub fn timespecff_diff(start: &mut TimeSpec, stop: &mut TimeSpec, result: &mut T
         *result = result_sec + result_nsec;
     }
 }
+
+fn open_shmem(shm_name: &str) -> *mut c_void {
+    let shm_fd = shm_open(
+        shm_name,
+        OFlag::O_CREAT | OFlag::O_RDWR,
+        Mode::S_IRWXU | Mode::S_IRWXG | Mode::S_IRWXO,
+    )
+    .unwrap();
+    let shm_ptr = unsafe {
+        ftruncate(shm_fd, SHM_SIZE as libc::off_t);
+        mmap(
+            None,
+            NonZeroUsize::new_unchecked(SHM_SIZE),
+            ProtFlags::PROT_READ | ProtFlags::PROT_WRITE,
+            MapFlags::MAP_SHARED,
+            shm_fd,
+            0,
+        )
+        .unwrap()
+    };
+    unsafe { msync(shm_ptr, SHM_SIZE, MS_SYNC) };
+
+    shm_ptr
+}
+
+fn read_shmem(shm_name: &str) -> Vec<u32> {
+    let shm_ptr = open_shmem(shm_name);
+
+    let mut vec_data: Vec<u32> = unsafe {
+        let data = slice::from_raw_parts(shm_ptr as *const u8, SHM_SIZE);
+        slice::from_raw_parts(data.to_vec().as_ptr() as *const u32, data.len() / 4).to_vec()
+    };
+    vec_data.retain(|&x| x != 0);
+    vec_data.retain(|x| kill(Pid::from_raw(*x as i32), None).is_ok());
+    unsafe { munmap(shm_ptr, SHM_SIZE).unwrap() };
+
+    vec_data
+}
+
+fn write_shmem(shm_name: &str, input: &Vec<u32>) {
+    let shm_ptr = open_shmem(shm_name);
+    let shm_byte = unsafe {
+        slice::from_raw_parts(input.as_ptr() as *const u8, size_of::<u32>() * input.len())
+    };
+    let addr = shm_ptr as *mut u8;
+    for (i, item) in shm_byte.iter().enumerate() {
+        unsafe { *addr.add(i) = *item };
+    }
+    unsafe { munmap(shm_ptr, SHM_SIZE).unwrap() };
+}
+
+fn lock_shmem(shm_fd: &i32) {
+    let lock = flock {
+        l_type: libc::F_WRLCK as i16,
+        l_whence: libc::SEEK_SET as i16,
+        l_start: 0,
+        l_len: 0,
+        l_pid: 0,
+    };
+    fcntl(*shm_fd, F_SETLKW(&lock)).unwrap();
+}
+
+fn unlock_shmem(shm_fd: &i32) {
+    let lock = flock {
+        l_type: libc::F_UNLCK as i16,
+        l_whence: libc::SEEK_SET as i16,
+        l_start: 0,
+        l_len: 0,
+        l_pid: 0,
+    };
+    fcntl(*shm_fd, F_SETLKW(&lock)).unwrap();
```

Comment:
This may fails.

Reply:
if say about fail to lock, that will blocking

Reply:
I mean, `unwrap` could be fail if something happens